src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOrganisationChecksOrchestrator.cs

```diff
@@ -227,5 +227,53 @@ public async Task<IcoNumberViewModel> GetIcoNumberViewModel(GetIcoNumberRequest
 
             return model;
         }
+
+        public async Task<WebsiteViewModel> GetWebsiteViewModel(GetWebsiteRequest request)
+        {
+            _logger.LogInformation($"Retrieving Website check details for application {request.ApplicationId}");
+
+            var pageId = GatewayPageIds.WebsiteAddress;
+
+            var commonDetails =
+                await _applyApiClient.GetPageCommonDetails(request.ApplicationId, pageId, request.UserName);
+
+            var model = new WebsiteViewModel
+            {
+                ApplicationId = request.ApplicationId,
+                PageId = pageId,
+                UkrlpLegalName = commonDetails.LegalName,
+                Ukprn = commonDetails.Ukprn,
+                Status = commonDetails.Status,
+                OptionPassText = commonDetails.OptionPassText,
+                OptionFailText = commonDetails.OptionFailText,
+                OptionInProgressText = commonDetails.OptionInProgressText,
+                SourcesCheckedOn = commonDetails.CheckedOn,
+                ApplicationSubmittedOn = commonDetails.ApplicationSubmittedOn,
+                Caption = RoatpGatewayConstants.Captions.OrganisationChecks,
+                Heading = RoatpGatewayConstants.Headings.Website
+            };
+
+            var ukrlpWebsite = await _applyApiClient.GetWebsiteAddressSourcedFromUkrlp(request.ApplicationId);
+            if (string.IsNullOrEmpty(ukrlpWebsite))
+            {
+                var applyWebsite = await _applyApiClient.GetWebsiteAddressManuallyEntered(request.ApplicationId);
+                if (!string.IsNullOrEmpty(applyWebsite))
+                {
+                    model.SubmittedWebsite = applyWebsite;
+                }
+            }
+            else
+            {
+                model.SubmittedWebsite = ukrlpWebsite;
+            }
```

Comment:
I'd have thought it would make sense for this logic to go in Apply so me can keep admin as dumb as possible.

Reply:
If moving this API-side, I'd recommend returning the source (UKRLP/Apply) in the data returned

Reply:
I'll move the logic to ApplyAPI, but why we need the source  (UKRLP/Apply) ? We don't make use of it. Please, confirm that it needs to be done @David-B-Read

Reply:
Logging API side to indicate where the website was retrieved from would be fine instead

Reply:
Sure, will do that. Thanks!